src/pages/components/ParticipantsForm.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+
+export default function ParticipantsForm({post}) {
+    const nameCount = 4; // default number of name fields
+    const [inputFields, setInputFields] = useState(
+        Array(nameCount).fill().map((_, i) => ({index: i, name: ''}))
+      )
+    
+    const handleFormChange = (index, event) => {
+        let data = [...inputFields];
+        data[index][event.target.name] = event.target.value;
+        setInputFields(data);
+    }
+    
+    const addField = () => {
+        const newField = {name: ''};
+        setInputFields([...inputFields, newField])
+    }
+    
+    const submit = (e) => {
+        e.preventDefault(); // prevents page refresh
+        console.log(inputFields);
+        post(inputFields.map(({index, name}) => name))
+    }
+
+    return (
+    <div>
+        <p className='recvField'>
+        <div className="App">
+              <form onSubmit={submit}>
```

Comment:
`Warning: <form> cannot appear as a descendant of <p>.` See explanation above

Reply:
removed paragraph